### PR TITLE
make subquery in SubqueryRouteTest work well

### PR DESCRIPTION
**Pull request by strongduanmu**
Ref #4696 .

Changes proposed in this pull request:
- modify getSubqueryWhereSegments method to make subquery in SubqueryRouteTest work well

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java [hunk 1]
```diff
@@ -108,15 +106,6 @@
      * @return subquery where segment collection.
      */
     public Collection<WhereSegment> getSubqueryWhereSegments() {
-        Collection<WhereSegment> subqueryWhereSegments = new ArrayList<>();
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java:111]: Hi, Sorry for the late feedback.
Do you think it is possible to move this function to `SelectStatementContext` referring to [#6319](https://github.com/apache/shardingsphere/pull/6319/files).
Basically, `xxxStatement` is designed as a POJO object, and we put more logic coding into `xxxStatementContext` for `rewriting`.

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java:111]: Besides, I presume #6319 finished part of your work, so my suggestion is you'd better give it check, and if possible, a coding merging and optimizing is welcomed. :-)

Comment (strongduanmu) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/statement/dml/SelectStatement.java:111]: @tristaZero Thank you for your suggestions, I will check and optimize this function. 😊

**Comment by tristaZero** [comment]
Hi @strongduanmu  Thanks for your continued efforts!
To make `SelectStatementContext` simple and clean, `xxxUtil` in a new `util` package is expected to consider. 
What do you think?

Besides, I also think  #6319  also needs a `xxxUtil`  to extract `all tables` from `SelectStatementContext`. Still, I am not sure it is a practical idea or not. So I'd glad to listen to your thought.  @jingshanglu  @strongduanmu 

Thanks,
Trista

**Comment by strongduanmu** [comment]
@tristaZero I agree with your idea. Extracting this function into util class will make SelectStatementContext look cleaner. I will optimize this pr later. 😊

**Pull request merged**